autoload/neomake.vim

```diff
@@ -368,9 +371,9 @@ function! s:AddExprCallback(maker) abort
 
     if list_modified
         if file_mode
-            call setloclist(a:maker.winnr, list, 'r')
+            call setloclist(a:maker.winnr, list, 'r', name)
```

Comment:
@nerdrew I think this is a pretty low-risk change, so if you can't get this condition to occur don't worry about it too much. However, in all the places you modified the call to `setloclist` or `setqflist` with what seems to be a neomake-only argument, I'd appreciate if you could make it ordinary-vim friendly. Unfortunately this means doing something like:

```
if has('nvim')
    " Neovim version
else
    " Vim version
endif
```

You're welcome to add functions wrapping this boilerplate in [autoload/neomake/utils.vim](https://github.com/benekastah/neomake/blob/master/autoload/neomake/utils.vim) if you think that's nicer.